src/merkle/merkle-tree.ts

```diff
@@ -83,6 +83,13 @@ export class MerkleTree<N, L extends N, M> {
       )
     }
 
+    // if there is only one leaf for the whole tree
+    if (elements.length === 1 && treeDepth === 0) {
+      const merged = await this.mergeFn.merge(elements[0], elements[0], treeMetadata)
```

Comment:
I don't think this can work, as the anchor commit includes the path through the merkle tree to the leaf node, so only one of the two paths would ever validate successfully.  I'd suggest using `null` for the right child here.

Reply:
no problem!